tests/regression/test_interpolate.py

```diff
@@ -358,3 +358,108 @@ def test_interpolate_periodic_coords_max():
     # All nodes on the "seam" end up being 1, not 0.
     assert np.allclose(np.unique(continuous.dat.data_ro),
                        [0.25, 0.5, 0.75, 1])
+
+
+# New FIAT Element
+
+from FIAT.dual_set import DualSet
+from FIAT.finite_element import CiarletElement
+from FIAT.reference_element import UFCInterval
+from FIAT.functional import PointEvaluation, IntegralMoment
+from FIAT.quadrature import make_quadrature
+from FIAT.polynomial_set import ONPolynomialSet
+ufcint = UFCInterval()
+
+
+class FIAT_C0ModifiedDualSet(DualSet):
+
+    def __init__(self, cell, order):
+        assert cell == ufcint and order == 3
+        entity_ids = {0: {0: [0], 1: [1]},
+                      1: {0: [2, 3]}}
+        vertnodes = [PointEvaluation(ufcint, xx)
+                     for xx in ufcint.vertices]
+        Q = make_quadrature(ufcint, 3)
+        # 1st integral moment node is integral(1*f(x)*dx)
+        ones = np.asarray([1.0 for x in Q.pts])
+        # 2nd integral moment node is integral(x*f(x)*dx)
+        xs = np.asarray([x for (x,) in Q.pts])
+        intnodes = [IntegralMoment(ufcint, Q, ones),
+                    IntegralMoment(ufcint, Q, xs)]
+        nodes = vertnodes + intnodes
+        super(FIAT_C0ModifiedDualSet, self).__init__(nodes, ufcint, entity_ids)
+
+
+class FIAT_C0Modified(CiarletElement):
+    r"""
+    An element, only defined for order 3 on interval cells, with 4 nodes:
+    ..math::
+     phi_0'(f) = f(0.)
+     phi_1'(f) = f(1.)
+     pfi_2'(f) = \int_{0}^{1} f(x) \,dx
+     pfi_3'(f) = \int_{0}^{1} x f(x) \,dx
+     """
+
+    def __init__(self, cell, order):
+        assert cell == ufcint and order == 3
+        poly_set = ONPolynomialSet(ufcint, 3)
+        dual_set = FIAT_C0ModifiedDualSet(ufcint, 3)
+        super(FIAT_C0Modified, self).__init__(poly_set, dual_set, 3, 0)
+
+
+def test_fiat_c0modified():
+    # can create
+    el = FIAT_C0Modified(UFCInterval(), 3)
+    # dual eval gives expected values from sum of point evaluations
+    fns = (lambda x: x[0], lambda x: x[0]**2)
+    expected = ([0, 1, 1/2, 1/3], [0, 1, 1/3, 1/4])
+    for i in range(len(fns)):
+        node_vals = []
+        for node in el.dual.nodes:
+            pt_dict = node.pt_dict
+            node_val = 0.0
+            for pt in pt_dict:
+                for (w, _) in pt_dict[pt]:
+                    node_val += w * fns[i](pt)
+            node_vals.append(node_val)
+        assert len(node_vals) == len(expected[i])
+        assert np.allclose(node_vals, expected[i])
+
+
+# FInAT Equivalent
+from finat.fiat_elements import ScalarFiatElement
+
+
+class FInAT_C0Modified(ScalarFiatElement):
+    def __init__(self, cell, degree):
+        super(FInAT_C0Modified, self).__init__(FIAT_C0Modified(cell, degree))
+
+
+# Add to TSFC Supported Elements
+import tsfc
+tsfc.finatinterface.supported_elements["C0 Modified"] = FInAT_C0Modified
```

Comment:
Ideally, you want to do this (and the UFL registration, too) in the `setUp` of the test, and then undo it in the `tearDown`, so that you only have the test element active for the duration of the test, rather than enabling it permanently from the moment this test file is loaded.

Not sure what the correct `pytest` syntax is for doing that, but you can probably figure it out.

Reply:
Use a pytest [yield fixture](https://docs.pytest.org/en/stable/fixture.html#teardown-cleanup-aka-fixture-finalization)

Reply:
Sadly there is no way for me to do the same thing with UFL - I can call `register_element` but there is no `deregister_element`. I'll need to manually remove the key from the dictionary `ufl.finitelement.elementlist.ufl_elements` which I don't believe is part of the public UFL interface.

Reply:
Though I suppose `tsfc.finatinterface.supported_elements` isn't parti of the TSFC public interface either

Reply:
Done - see 6fc2de8286cd951e46dc68d9172d342562d75d3f